packages/core/src/build/configAndIndexingFunctions.ts

```diff
@@ -119,16 +119,17 @@ export async function buildConfigAndIndexingFunctions({
         "id" in c ? c.id === chain.id : false,
       );
 
-      // if (matchedChain === undefined) {
-      //   throw new Error(
-      //     `Chain "${chainName}" with id ${chain.id} not found in viem. Please update viem to the latest version.`,
-      //   );
-      // }
       if (chain.rpc === undefined) {
-        throw new Error(
-          `Chain "${chainName}" with id ${chain.id} has no RPC URL.`,
-        );
-      } else if (typeof chain.rpc === "string" || Array.isArray(chain.rpc)) {
+        if (matchedChain === undefined) {
+          throw new Error(
+            `Chain "${chainName}" with id ${chain.id} has no RPC URL.`,
```

Comment:
We could probably make this more useful, eg "has no RPC defined and no default RPC URL was found in viem/chains"